src/Core/Services/DeviceTrustCryptoService.cs

```diff
@@ -99,5 +99,33 @@ public async Task<DeviceResponse> TrustDeviceIfNeededAsync()
             await SetShouldTrustDeviceAsync(false);
             return response;
         }
+
+        public async Task<bool> IsDeviceTrustedAsync()
+        {
+            var existingDeviceKey = await GetDeviceKeyAsync();
+            return existingDeviceKey != null;
+        }
+
+        public async Task<UserKey> DecryptUserKeyWithDeviceKeyAsync(string encryptedDevicePrivateKey, string encryptedUserKey)
+        {
+            // Get device key
+            var existingDeviceKey = await GetDeviceKeyAsync();
+
+            if (existingDeviceKey == null)
+            {
+                // User doesn't have a device key anymore so device is untrusted
+                return null;
+            }
+
+            // Attempt to decrypt encryptedDevicePrivateKey with device key
+            var devicePrivateKey = await _cryptoService.DecryptToBytesAsync(
+              new EncString(encryptedDevicePrivateKey),
+              existingDeviceKey
+            );
+
+            // Attempt to decrypt encryptedUserDataKey with devicePrivateKey
+            var userKey = await _cryptoService.RsaDecryptAsync(encryptedUserKey, devicePrivateKey);
+            return new UserKey(userKey);
```

Comment:
⛏️ I think that given having objects that represent keys like `UserKey` we should use more specific variable names so we know what they represent (actual key object, bytes, encrypted string):
```suggestion
            // Attempt to decrypt encryptedDevicePrivateKey with device key
            var devicePrivateKeyBytes = await _cryptoService.DecryptToBytesAsync(
              new EncString(encryptedDevicePrivateKey),
              existingDeviceKey
            );

            // Attempt to decrypt encryptedUserDataKey with devicePrivateKey
            var userKeyBytes = await _cryptoService.RsaDecryptAsync(encryptedUserKey, devicePrivateKeyBytes);
            return new UserKey(userKeyBytes);
```